### ISSUE TITLE
Why assign eps as volume if cell is zero matrix?

### ISSUE THREAD

**Issue by CompRhys** [MDIL-SNU/SevenNet#212]
In this commit https://github.com/MDIL-SNU/SevenNet/commit/13347afa77ff1ccfe4e6fd46ba0fe5c55e9dccd9 the volume is set to eps if the cell is 0. What is the reason to do this? This is what causes the error in https://github.com/Radical-AI/torch-sim/pull/190 for benzene. If I get the volume in the SevenNetModel to eps the values are not equal due to numerical instability. If left as zero we get a collection of -inf and nan that are "equal". 

Is this needed? other models seemingly do not have this "fix".

**Comment by YutackPark** [comment]
We wanted to simultaneously train both molecule and bulk, without introducing data dependent branch in force output module (which is critical for torchscript). As molecule has no volume, I had to assign some value for it and 'eps' is chosen. 

I thought assigning '0' is more risky than eps, but it is not tested. If '0' works fine and is more advantageous, there is no reason to keep eps.

**Comment by CompRhys** [comment]
If the training code is already ignoring the stress loss for molecules then I think it's fine to leave as zero? For the only example of benzene I looked at there's no issue with it and it returns sensible inf/nan where you would expect inf in x-y plane and nan along z

I should say that it's a bit strange to me that the volume is an input at all rather than taken dynamically from the determinant of the cells.

**Comment by YutackPark** [comment]
> I should say that it's a bit strange to me that the volume is an input at all rather than taken dynamically from the determinant of the cells.

It (`atoms.cell.volume`) is the same, we're simply relying on ASE.
```python
@property
    def volume(self) -> float:
        """Get the volume of this cell.

        If there are less than 3 lattice vectors, return 0."""
        # Fail or 0 for <3D cells?
        # Definitely 0 since this is currently a property.
        # I think normally it is more convenient just to get zero
        return np.abs(np.linalg.det(self))
```

>If the training code is already ignoring the stress loss for molecules then I think it's fine to leave as zero? For the only example of benzene I looked at there's no issue with it and it returns sensible inf/nan where you would expect inf in x-y plane and nan along z

I agree. Actually, thanks for the check and report. Please check #213
I also removed redundant `CELL` and `CELL_SHIFT` fields of dataloding. I think 7net model in TorchSim can safely remove those fields.

**Comment by CompRhys** [comment]
> I also removed redundant `CELL` and `CELL_SHIFT` fields of dataloding. I think 7net model in TorchSim can safely remove those fields.

This confuses me as I think both `CELL` and `CELL_SHIFT` are used in the code? https://github.com/MDIL-SNU/SevenNet/blob/24ebbe223ebf252a7e3d148f0afd266ce03d7690/sevenn/nn/edge_embedding.py#L25-L31

**Comment by YutackPark** [comment]
Yes they was used in the code. Especially from `EdgePreprocess` and `ForceStressOutput` modules. They used to compute stress by autograd and an auxiliary `strain` tensor, which is more common way to computing stress tensor in GNN based interatomic potentials.

However, the same stress can be obtained from edge-wise forces. If you interested, check below module. (code is not kind tough)
https://github.com/MDIL-SNU/SevenNet/blob/24ebbe223ebf252a7e3d148f0afd266ce03d7690/sevenn/nn/force_output.py#L140

7net has migrated to this method. This method can also be used in the LAMMPS parallel version, while the `strain` version can not. This is a tricky part that is hard to find in the literature.
